ipv8/configuration_pydantic.py

```diff
@@ -0,0 +1,402 @@
+"""
+Pydantic configuration BaseModels for IPv8.
+
+If you want to see the available configuration options, print the schema as follows:
+
+::
+
+        from json import dumps
+        print(dumps(format_schema_recursive(IPv8Configuration), indent=4))
+
+|
+You can use the ``IPv8Configuration by simply passing it to an ``IPv8`` constructor:
+
+::
+
+        ipv8_instance = IPv8(IPv8Configuration())
+
+"""
+
+import base64
+import inspect
+import sys
+from enum import Enum
+from typing import Any, Dict, List, Optional, Set, Type, Union
+
+from pydantic import BaseModel, Field, validator
+from pydantic.validators import dict_validator
+
+from .configuration import DISPERSY_BOOTSTRAPPER, default
+from .keyvault.crypto import default_eccrypto
+
+
+class AbstractModel:
```

Comment:
The problem: "`AbstractModel` is too restrictive to the end users (developers)."

In the thread below we came up with two possible examples of config usage

1: 
```python
class MyWalkerConfig(Walker):
    strategy = 'MyWalker'
    init = {'foo': 13}

walkers = [
    MyWalkerConfig()
]
```

2:
```python
walkers = [
    Walker(strategy='MyWalker', init={'foo': 13})
]
```

As an end user, I would like to have both options of using the `IPv8` configuration. 
Therefore I can decide which approach suits my application better. 
`AbstractModel` forces me to use only (1) approach.